File: subroutines/selected/dependencies/subroutine prokinkcalcla dependencies/function dacoscheck.f
      function dacoscheck(dargin,argin,idbp,iout,lab)
c*****Make sure cos is in the [-1,1] range, return dacos(arg)
      real*8 dargin,arg
      character*(*) lab
      if (idbp .eq. 1) then
        arg=dargin
      else
        arg=argin
      end if
      if (arg .lt. -1.0d0) then
        if (arg .ge. -1.01d0) then
          arg=-1.0d0
          dacoscheck=dacos(-1.0d0)
          return
        else
          write (iout,1000) arg,lab
          dacoscheck=0.d0
        end if
      else if (arg .gt. 1.0d0) then
        if (arg .le. 1.01d0) then
          arg=1.0d0
          dacoscheck=dacos(+1.0d0)
          return
        else
          write (iout,1000) arg,lab
          dacoscheck=dacos(+1.0d0)
        end if
      else
        dacoscheck=dacos(arg)
        return
      end if
      return
1000  format(' PROGRAM ERROR: invalid sin or cos value:',f10.6,
     -  ' in subroutine ',a)
      end

File: fortran to python conversion/dependencies/function dacoscheck.py
import math
import numpy as np

def dacoscheck(arg):
	"""
	Make sure cos is in the [-1,1] range, return acos(arg)
	otherwise raise an exception
	***
	Parameters:
	dargin: input angle
	argin:
	idbp:

	Returns:
	acos(dargin) if idb=1 else acos(argin)
	"""
	#if (idb == 1):
		#arg = dargin
	#else:
		#arg = argin
	#Checking arg value and returning acos(arg) if correct
	if ((arg > 1 and arg<=1.01) or (arg < -1 and arg >=-1.01)):
		# arg is close enough to the [-1,1] range to be approximated by its boundaries
		arg = np.sign(arg)
	elif( arg >1.01 or arg < -1.01):
		raise ValueError("PROGRAM ERROR: invalid sin or cos value: " +arg)
	return math.acos(arg)